File: code/ClassMethodCall.php
<?php

class AnimalH
{
    public function walk()
    {
        $this->bark();
        $this->guard();
        echo "I'm walking\n";
    }

    public function bark()
    {
        echo "Wuf!\n";
    }

    private function guard()
    {
        echo "WUUUF!\n";
    }
}


(new AnimalH())->walk();

File: code/java/ClassMethodCall.java
class Animal {
  public void walk() {
    bark();
    guard();
    System.out.println("I'm walking");
  }

  public void bark() {
    System.out.println("Wuf!");
  }

  private void guard() {
    System.out.println("WUUUF!");
  }
}

class ClassMethodCall {
  public static void main(String[] args) {
    new Animal().walk();
  }
}
